bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -0,0 +1,290 @@
+            + "=<bigtable-instance-id>\n"
+            + "Required arguments:\n"
+            + "  snapshot-name     - name of hbase snapshot to read\n"
+            + "  snapshot-dir      - directory of snapshot, e.g., gs://hbase-migration-table1-bucket/export/table1-snapshot\n"
+            + "  tablename         - bigtable table to import into\n"
+            + "  temp-dir          - a temporary base directory to restore hlinks of the snapshot into for read, e.g., gs://hbase-migration-table1-bucket/export/table1-restore\n");
+
+    System.exit(1);
+  }
+
+  /**
+   * Job runner
+   *
+   * @param args command line input
+   * @return job success flag
+   * @throws Exception terminal for application failure
+   */
+  @Override
+  public int run(String[] args) throws Exception {
+    Configuration conf = HBaseConfiguration.create(getConf());
+    String[] otherArgs = new GenericOptionsParser(conf, args).getRemainingArgs();
+
+    if (setConfFromArgs(conf, otherArgs) != 0) return -1;
+
+    LOG.info(
+        "Using Arguments: "
+            + "\n"
+            + "snapshotName = "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + "\n"
+            + "snapshotDir = "
+            + conf.get(HConstants.HBASE_DIR)
+            + "\n"
+            + "bigtableName = "
+            + conf.get(TableOutputFormat.OUTPUT_TABLE)
+            + "\n"
+            + "tempDir = "
+            + conf.get(SNAPSHOT_RESTOREDIR_KEY)
+            + "\n"
+            + CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY
+            + " = "
+            + conf.get(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY)
+            + "\n");
+
+    Job job = createSubmittableJob(conf);
+    int successStatus = (job.waitForCompletion(true) ? 0 : 1);
+
+    Counters counters = job.getCounters();
+    long numRows = counters.findCounter(ScanCounter.NUM_ROWS).getValue();
+    long numCells = counters.findCounter(ScanCounter.NUM_CELLS).getValue();
+
+    System.out.println(("############# Num of ROWS imported= " + numRows));
+    System.out.println(("############# Num of CELLS imported= " + numCells));
+
+    return successStatus;
+  }
+
+  protected static int setConfFromArgs(Configuration conf, String[] args) {
+    // basic pre-condition check
+    if (args.length < 4) {
+      usage("Wrong number of arguments: " + args.length);
+      return -1;
+    }
+    if (conf.get(BigtableOptionsFactory.PROJECT_ID_KEY) == null) {
+      usage("Must specify the property " + BigtableOptionsFactory.PROJECT_ID_KEY);
+      return -1;
+    }
+    if (conf.get(BigtableOptionsFactory.INSTANCE_ID_KEY) == null) {
+      usage("Must specify the property " + BigtableOptionsFactory.INSTANCE_ID_KEY);
+      return -1;
+    }
+
+    // set arguments in configuration for ease of use
+    conf.set(SNAPSHOTNAME_KEY, args[0]);
+    conf.set(HConstants.HBASE_DIR, args[1]);
+    conf.set(TABLENAME_KEY, args[2]);
+    conf.set(SNAPSHOT_RESTOREDIR_KEY, args[3]);
```

Comment:
Do these not need precondition checks as well?

Reply:
we check the number of args as a precondition for basic verification of args such as the other mr jobs do.